src/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Activity.Id.Random.cs

```diff
@@ -3,13 +3,13 @@
     public partial class Activity
     {
         #region private
-        private string generateUniquePrefix()
-        {
-            byte[] bytes = new byte[8];
-            s_random.Value.NextBytes(bytes);
 
-            return $"{s_rootIdPrefix}{BitConverter.ToUInt64(bytes, 0):x}";
+        private string GenerateInstancePrefix()
+        {
+            byte[] bytes = Guid.NewGuid().ToByteArray();
```

Comment:
Make a routine long 'GetRandomNumber() that you call in the three places in the code where you call NewGuid(), since exactly how we create this random number is likely to change.   

My preference is that the implementation of this would be unsafe and simply do

Guid g = Guid.NewGuid();
return *((long*)  &g);

But since this is not on the 'hot' path' I am OK with your slower (but not unsafe) implementation.